app/Models/Article.php

```diff
@@ -138,21 +138,6 @@ public function scopeSortByPublishedDate(Builder $query, string $direction = 'de
         return $query->orderByRaw("articles.published_at {$direction} {$nullsPosition}");
     }
 
-    /**
-     * @param  Builder<self>  $query
-     * @return Builder<self>
-     */
-    public function scopeWithFeaturedCollections(Builder $query): Builder
-    {
-        return $query->with(['collections' => function ($query) {
-            $query->select([
-                'collections.name',
-                'collections.slug',
-                'collections.extra_attributes->image as image',
-            ]);
-        }]);
```

Comment:
this scope used to select only those columns, what if rename the scope to `withRelatedCollections` or something like that so we dont lose the select

Reply:
I can add that, i just didn't see any value in selecting only few columns - in fact i see something that's bothering me, and that's the fact that we're selecting `extra_attributes->image` as `image`, and then use `$model->image` to reference that "imaginary" field

I don't think these imaginary attributes are a good approach, especially with the lack of type safety with Laravel

Reply:
> i just didn't see any value in selecting only few columns
Why? You would like to select all columns when you don't need?

Reply:
I don't see why not, what is the issue with selecting all columns in this case?

Reply:
It is not a good approach to select everything when you don't need:

![image](https://github.com/ArdentHQ/dashbrd/assets/132887516/dde8baa9-2df4-4f86-9f15-b7dc4e9af617)

Reply:
Sure, some random page on the internet says that. In practice and in this concrete example, why is bad to select all columns for a table with 1000 records and 25 columns? Don't you think that having some virtual, imaginary property on a model is objectively worse than selecting all columns from such a tiny table? Let's not make the code less readable and less maintainable for the sake of literal micro-optimizations.

I can give at least 3 examples related to database optimizations that are more important than this change, yet I didn't see you commenting on that. Let's not nitpick.

Reply:
> Sure, some random page on the internet says that. In practice and in this concrete example, why is bad to select all columns for a table with 1000 records and 25 columns?

I can give you more credible source if you want, I've read it many times:
High Performance MySQL: Optimization, Backups, and Replication Third Edition: [link](https://gateway.ipfs.io/ipfs/bafykbzacebhf4eml4bpzymd7zwoothcyr6j54mwvez7k335abio32cl7ewklm?filename=Baron%20Schwartz%2C%20Peter%20Zaitsev%2C%20Vadim%20Tkachenko%20-%20High%20performance%20MySQL_%20optimization%2C%20backups%2C%20and%20replication-O%E2%80%99Reilly%20Media%20%282012%29.pdf)
Chapter 6: Query Performance Optimization page 203
![image](https://github.com/ArdentHQ/dashbrd/assets/132887516/4a16e896-2ff3-4324-bff8-3680f1e6509b)


> Don't you think that having some virtual, imaginary property on a model is objectively worse than selecting all columns from such a tiny table?

That is a different problem than above, why would follow good approaches in one direction but not in another, if good practices say not to use `virtual` columns, using a proper name would solve this.

I am not nitpick-ing, if we are following good practices, we should do it when possible. No hard feelings.

Reply:
I was more talking about the fact that just because it's said somewhere doesn't mean you have to blindly follow it for the sake of just "doing best practices" for truly negligible performance improvement of 10-15 milliseconds when it can also be handled with caching -- especially in the early MVP stage where tables are modified a bunch and we have to run back every 5th PR to change which columns are selected where.

Also, there's no need to explain that selecting all columns is performing worse than selecting a few columns -- judging from the dozen of my PRs that heavily optimized SQL queries in this project, I'm very sure I don't need to be educated on how to optimize the queries that matter.

And yes, this was nitpicking.

You're talking about two separate problems, which are indeed there. My "Don't you think" was about the fact that you didn't nitpick in such a way when this "aliasing and then assigning to a magic class property" was introduced. Please, let's focus on more important things.

Reply:
@alfonsobries will push your suggestions